pkg/grpc/universalapi/api_subtlecrypto_subtlecrypto.go

```diff
@@ -0,0 +1,351 @@
+//go:build subtlecrypto
+// +build subtlecrypto
+
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package universalapi
+
+import (
+	"context"
+	"crypto"
+	"crypto/x509"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"time"
+
+	"github.com/lestrrat-go/jwx/v2/jwk"
+
+	contribCrypto "github.com/dapr/components-contrib/crypto"
+	diag "github.com/dapr/dapr/pkg/diagnostics"
+	"github.com/dapr/dapr/pkg/messages"
+	runtimev1pb "github.com/dapr/dapr/pkg/proto/runtime/v1"
+	"github.com/dapr/dapr/pkg/resiliency"
+	kitCrypto "github.com/dapr/kit/crypto"
+)
+
+// SubtleGetKeyAlpha1 returns the public part of an asymmetric key stored in the vault.
+func (a *UniversalAPI) SubtleGetKeyAlpha1(ctx context.Context, in *runtimev1pb.SubtleGetKeyAlpha1Request) (*runtimev1pb.SubtleGetKeyAlpha1Response, error) {
+	component, err := a.cryptoValidateRequest(in.ComponentName)
+	if err != nil {
+		return &runtimev1pb.SubtleGetKeyAlpha1Response{}, err
+	}
+
+	// Get the key
+	policyRunner := resiliency.NewRunner[jwk.Key](ctx,
+		a.Resiliency.ComponentOutboundPolicy(in.ComponentName, resiliency.Crypto),
+	)
+	start := time.Now()
+	res, err := policyRunner(func(ctx context.Context) (jwk.Key, error) {
+		return component.GetKey(ctx, in.Name)
+	})
+	elapsed := diag.ElapsedSince(start)
+
+	diag.DefaultComponentMonitoring.CryptoInvoked(ctx, in.ComponentName, diag.Get, err == nil, elapsed)
```

Comment:
diagnostics wise, the key is got here ... but then it can fail to be marshalled leading to an error, shouldn't diagnostics be set only after it is marshalled, just be returning the value?

Reply:
Looks kike this is consistent with what we do for other building blocks: https://github.com/ItalyPaleAle/dapr/blob/e07039b316f2ff8cfc0558d42c73e9054a539789/pkg/grpc/api.go

I think this is fine as what we want to measure here is the success of the operation from the component and the latency it adds